org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -240,10 +239,12 @@ private String create() throws Exception {
 						//	Optional Restrictions
 						+ " AND (dl.Group1 IS NULL OR p.Group1=dl.Group1)"
 						+ " AND (dl.Group2 IS NULL OR p.Group2=dl.Group2)"
-						+ " AND (dl.C_BPartner_ID IS NULL OR EXISTS "
-						+ "(SELECT * FROM M_Product_PO po "
-						+ "WHERE po.M_Product_ID=p.M_Product_ID AND po.C_BPartner_ID=dl.C_BPartner_ID))"
-						+ " AND (dl.M_Product_ID IS NULL OR p.M_Product_ID=dl.M_Product_ID)");
+						+ " AND (dl.C_BPartner_ID IS NULL AND dl.VendorCategory IS NULL OR EXISTS "

```

Comment:
The change here looks wrong:
from
dl.C_BPartner_ID IS NULL OR EXISTS
to
dl.C_BPartner_ID IS NULL AND dl.VendorCategory IS NULL OR EXISTS

The AND takes precedence, if is intentional better to make it explicit with parenthesis, but I think is not, what if a record has set the BP but not the category?  or viceversa?

Reply:
Ok, yet another point where I should include brackets. It's intentional to take precedence. Former only if a bpartner was present, existence of a M_Product_PO record must be ensured. Now it must be ensured also if VendorCategory is not null. And of course only the not null part then must be used to decide whether an appropriate M_Product_PO record is present.